capten/agent/pkg/capten-store/app_config_store.go

```diff
@@ -5,148 +5,222 @@ import (
 )
 
+func init() {
+	p := make([]string, len(fields))
+	for i := 0; i < len(fields); i++ {
+		p[i] = "?"
+	}
+
+	placeholder := strings.Join(p, ", ")
+
+	InsertAllAppConfigQuery = "INSERT INTO apps.AppConfig( " +
+		strings.Join(fields, ", ") +
+		fmt.Sprintf(" ) VALUES (%s)", placeholder)
+
+}
+
+func CreateSelectByFieldNameQuery(field string) string {
+	return "SELECT" +
+		strings.Join(fields, ", ") +
+		"FROM apps.AppConfig" + fmt.Sprintf(" WHERE %s = ?", field)
+}
+
 const (
-	AppConfigTable string = "AppConfig"
-
-	insertAppConfigQuery string = `
-		INSERT INTO %s.%s(
-			name, chart_name, repo_name, repo_url,
-			namespace, release_name, version, override,
-			create_namespace, privileged_namespace
-		) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`
-
-	selectAppConfigByNameQuery string = `
-		SELECT name, chart_name, repo_name, repo_url, 
-				namespace, release_name, version, override,
-				create_namespace, privileged_namespace
-		FROM %s.%s WHERE name = ? LIMIT 1`
-
-	updateAppConfigByNameQuery string = `
-		UPDATE %s.%s SET %s WHERE name = ?`
+	appName, description, category       = "app_name", "description", "category"
+	chartName, repoName, repoUrl         = "chart_name", "repo_name", "repo_url"
+	namespace, releaseName, version      = "namespace", "release_name", "version"
+	launchUrl, launchRedirectUrl         = "launch_url", "launch_redirect_url"
+	createNamespace, privilegedNamespace = "create_namespace", "privileged_namespace"
+	overrideValues, launchUiValues       = "override_values", "launch_ui_values"
 )
 
-func (a *Store) AddAppConfig(config types.AppConfig) error {
-	query := a.client.Session().Query(fmt.Sprintf(insertAppConfigQuery, a.conf.Keyspace, AppConfigTable))
-	var override []byte
-	var err error
-	if config.Override != nil {
-		override, err = yaml.Marshal(config.Override)
-		if err != nil {
-			return err
-		}
+var (
+	fields []string = []string{
+		appName, description, category,
+		chartName, repoName, repoUrl,
+		namespace, releaseName, version,
+		launchUrl, launchRedirectUrl,
+		createNamespace, privilegedNamespace,
+		overrideValues, launchUiValues,
+	}
+
+	InsertAllAppConfigQuery           string
+	UpdateAppConfigByReleaseNameQuery = "UPDATE apps.AppConfig SET %s WHERE release_name = ?"
+)
+
+func (a *Store) AddAppConfig(config *agentpb.SyncAppData) error {
+	if config.Config == nil {
+		return fmt.Errorf("config value nil")
 	}
+	query := a.client.Session().Query(InsertAllAppConfigQuery)
+
+	override, launchUiValues := "", ""
+	if config.Values != nil {
+		override, launchUiValues =
+			string(config.Values.OverrideValues), string(config.Values.LaunchUIValues)
+	}
+	c := config.Config
 	return query.Bind(
-		config.Name,
-		config.ChartName,
-		config.RepoName,
-		config.RepoURL,
-		config.Namespace,
-		config.ReleaseName,
-		config.Version,
-		string(override),
-		config.CreateNamespace,
-		config.PrivilegedNamespace,
+		c.AppName, c.Description, c.Category,
+
```

Comment:
all attributes defined in the proto files are not added to DB.

icon and installStatus missing

icon expected to store bytes with BLOB

Reply:
done